pkg/frontend/querymiddleware/codec.go

```diff
@@ -255,6 +255,8 @@ func (c prometheusCodec) decodeInstantQueryRequest(r *http.Request) (Request, er
 	return &result, nil
 }
 
+// DecodeRangeQueryTimeParams encapsulates Prometheus instant query time param parsing,
+// emulating the logic in prometheus/prometheus/web/api/v1#API.query_range.
```

Comment:
I don't feel particularly strongly about this, but what do you think about a direct link?
```suggestion
// emulating the logic in https://github.com/prometheus/prometheus/blob/0763ec841bdc75d163a123dbffd05b12ff0500f1/web/api/v1/api.go#L505-L582.
```

Reply:
(same for the other ones)